triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunner.scala

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.engine.trigger
+
+import akka.actor.typed.{Behavior, PostStop}
+import akka.actor.typed.scaladsl.AbstractBehavior
+import akka.actor.typed.SupervisorStrategy._
+import akka.actor.typed.Signal
+import akka.actor.typed.PostStop
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.ActorContext
+import akka.stream.Materializer
+import com.typesafe.scalalogging.StrictLogging
+import com.daml.grpc.adapter.ExecutionSequencerFactory
+
+object TriggerRunner {
+  type Config = TriggerRunnerImpl.Config
+
+  trait Message
+  final case object Stop extends Message
+
+  def apply(config: Config, name: String)(
+      implicit esf: ExecutionSequencerFactory,
+      mat: Materializer): Behavior[TriggerRunner.Message] =
+    Behaviors.setup(ctx => new TriggerRunner(ctx, config, name))
+}
+
+class TriggerRunner(
```

Comment:
What’s the reason for this being a class instead of an `Object` like `TriggerRunnerImpl`?

Reply:
`TriggerRunner` maintains state in the form a data member, `TriggerRunnerImpl` passes immutable state around as parameters. It's the difference between the functional vs. the object oriented style. I fully expect the former could be rewritten in the functional approach but in this instance it seemed easier to just go with the OO style (consistent with this example https://doc.akka.io/docs/akka/current/typed/guide/tutorial_1.html#failure-handling).